File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/collection_storage_provider.java

package com.gracenote.gnsdk;

public class collection_storage_provider {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected collection_storage_provider(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(collection_storage_provider obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_collection_storage_provider(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public collection_storage_provider() {
    this(gnsdk_javaJNI.new_collection_storage_provider__SWIG_0(), true);
  }

  public collection_storage_provider(collection_storage_provider arg0) {
    this(gnsdk_javaJNI.new_collection_storage_provider__SWIG_1(collection_storage_provider.getCPtr(arg0), arg0), true);
  }

  public String getData(long pos) {
    return gnsdk_javaJNI.collection_storage_provider_getData(swigCPtr, this, pos);
  }

  public long count() {
    return gnsdk_javaJNI.collection_storage_provider_count(swigCPtr, this);
  }

}

File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/collection_storage_provider.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

public class collection_storage_provider : IDisposable {
  private HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal collection_storage_provider(IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(collection_storage_provider obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~collection_storage_provider() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_collection_storage_provider(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
    }
  }

  public collection_storage_provider() : this(gnsdk_csharp_marshalPINVOKE.new_collection_storage_provider__SWIG_0(), true) {
  }

  public collection_storage_provider(collection_storage_provider arg0) : this(gnsdk_csharp_marshalPINVOKE.new_collection_storage_provider__SWIG_1(collection_storage_provider.getCPtr(arg0)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public string get_data(uint pos) {
    string ret = gnsdk_csharp_marshalPINVOKE.collection_storage_provider_get_data(swigCPtr, pos);
    return ret;
  }

  public uint count() {
    uint ret = gnsdk_csharp_marshalPINVOKE.collection_storage_provider_count(swigCPtr);
    return ret;
  }

}

}
